homeassistant/components/recorder/statistics.py

```diff
@@ -1953,11 +1981,7 @@ def _sorted_statistics_to_dict(
     metadata = dict(_metadata.values())
     need_stat_at_start_time: set[int] = set()
     stats_at_start_time = {}
-
-    def no_conversion(val: float | None) -> float | None:
-        """Return val."""
-        return val
-
+    start_time_ts = start_time.timestamp() if start_time else None
     # Set all statistic IDs to empty lists in result set to maintain the order
```

Comment:
We can move this later and check if we say it before creating the entry in the dict. We can also get rid of the default dict.  Than we don't have to filter it at the end.

Reply:
No luck getting rid of the defaultdict, but I did get rid of the extra filtering